user/src/com/google/gwt/i18n/rebind/MessagesInterfaceCreator.java

```diff
@@ -100,4 +136,96 @@ protected String javaDocComment(String path) {
     return "Interface to represent the messages contained in resource bundle:\n\t"
       + path + "'.";
   }
+  
+  private void determineNumberType(String subFormat) {
+    switch (subFormat) {
+      case "integer":
+        composer.print("Integer");
+        break;
+      case "currency":
+      case "percent":
+      default:
+        if(subFormat.contains(".")) {
+          composer.print("Double");
+        } else {
+          composer.print("Integer");
+        }
+    }
+  }
+
+  private String determineReturnType(Map<Integer, ArgumentChunk> args) {
+    for (ArgumentChunk arg : args.values()) {
+      if ("safehtml".equals(arg.getFormat())) {
+        return "com.google.gwt.safehtml.shared.SafeHtml";
+      }
+    }
+    return "String";
+  }
+
+  private void genPluralsAnnotation(Map<String, String> plurals) {
+    composer.print("@AlternateMessage({");
+    String[] keys = plurals.keySet().toArray(new String[] {});
+    if (keys.length > 1) {
+      composer.println("");
+      composer.indent();
+    }
+    for (int i = 0; i < keys.length; i++) {
+      String key = keys[i];
+      if (i > 0) {
+        composer.println(",");
+      }
+      composer.print("\"" + key.substring(key.indexOf('[') + 1, key.length() - 1) + "\", ");
+      composer.print("\"" + plurals.get(key) + "\"");
```

Comment:
Probably should use makeJavaString here - in theory the keys never have strings in them, but I _think_ it is technically allowed for properties files to have spaces in their keys if escaped?